### PR TITLE
Add resource index

### PR DESCRIPTION
**Pull request by ronanyeah**
I've rounded up all the workshops and morning challenges for easier discovery as time goes on, and to keep track of resources as topics are retired from the curriculum, such as Hapi. Some thoughts:

__Naming:__
Tracking them down was a pain due to the inconsistent repo naming (which I brought up before [here](https://github.com/foundersandcoders/master-reference/issues/323)). I'd like to rename many of them for consistency. Is there a 'making a workshop' checklist already? If not I can make one. For example:
1. Have a learning outcomes
2. Follow repo naming convention
3. Add to resource index

__Duplication:__
There are several overlapping repos on topics such as express and postgres, so hopefully having a central list could help with this, and steer people towards collaboration rather than yet another new repo.

__Modularity:__
Our workshops are usually coupled quite tightly to a certain week of the curriculum, but I think we could be a lot more flexible with the morning challenges. Perhaps have a 'pool' of them that can be used in any week depending on the needs and interests of that particular cohort, and even be attempted by students on their own at any point during the course, or afterwards.

__Additions:__
There's probably a lot more FAC content in personal GH repos so it might be worth putting the word out to track them down, and perhaps fork them to the FAC org. (Surely we have more morning challenges than this??)

### CODE REVIEW

**Comment by ronanyeah** [comment]
(Also this is just a list of links but it would probably be better to have a table with columns for what week they are usually in the curriculum, if appropriate etc)

**Comment by jsms90** [comment]
@ronanyeah wanna delete the master reference [`README.md`](https://github.com/foundersandcoders/master-reference/blob/master/README.md)?

**Comment by ronanyeah** [comment]
@jsms90 Yeah I think so, I was looking at that actually and found it a bit confusing. I think we could delete most of it (apart from the contrib instructions) and add the per week 'open issues' links to [this file](https://github.com/foundersandcoders/master-reference/tree/master/coursebook). (Also looks like `gitter_channels.md` should be moved to the 'playbook'?)

**Comment by eliascodes** [comment]
Are there morning challenges missing from this? There's like two per week (i.e. should be 16 total), but only 11 on the list.

**Comment by ronanyeah** [comment]
@eliascodes These are only the morning challenges that have their own repos (and may not even be currently on the curriculum), whereas a lot of morning challenges are being kept as `.md` files in the 'week' folders. Now I don't think we need to start giving all morning challenges a repo, so maybe we should think about keeping the `.md` ones in their own folder, and they can be linked to from the index, and from the 'week' timetable.

**Comment by jsms90** [comment]
@ronanyeah Only thing I prefer about the existing list is seeing all the workshops in the order in which we deliver them e.g. all the testing week workshops next to each other.

How about:

| Week | Workshops |
|---- |----|
|1 | Git |
|   | Accessibility |
| 2 | TDD code along |
|  | OTher thing|

and then, like you said, just have all the morning challenges in a list, separate from specific weeks.

**Comment by jsms90** [comment]
@des-des nah, the `README.md` contains most of the workshops & morning challenges that were used for the FAC11 iteration (except express week) and everything that was used for the FAC10 version. Its purpose was the same as what resource index is now trying to achieve, which is why it still contains workshops from past weeks e.g. hapi & oauth.

I'm not sure whether there is much value in having 2 files with a long list of workshops.
> the readme should contain a list of resources in the curriculum

Given that the vision of the master reference is for each week to be composable, would it not be better to have these lists sitting within the `week-x` folders? After all, the list is only really going to get used by the WFs and soon-to-be mentors, who don't really have time to review every single workshop in the curriculum.

> Our workshops are usually coupled quite tightly to a certain week of the curriculum, but I think we could be a lot more flexible with the morning challenges.

I would say that this is best achieved by having:
- a list of all morning challenges in the `general` folder i.e. this "resource index"
- a list of workshops within the `week-x` folders (with those workshops that are currently in use written in bold?)

`workshop.md`:
- **wc-node-shell**
- **wc-node-girls**
- wc-debugging
- another old one

**Comment by iteles** [comment]
@ronanyeah Do you want to do anything else on this?
Trying to figure out whether you'd like a review here now or later 😊

**Comment by ronanyeah** [comment]
@iteles There's a bit more to do on it I think, I'll have a look during the week.

**Comment by iteles** [comment]
Sounds good @ronanyeah, assign me when ready 👍

**Comment by ronanyeah** [comment]
Will get this finished tonight.

**Comment by ronanyeah** [comment]
Urgh conflict, will fix it in a couple of hours.

**Comment by m4v15** [comment]
hey @ronanyeah just did a tiny fix, hope thats alright/what you intended

also yeah I think the morning challenges that are md files in the MR would be better moved to a folder or something, might do it this week if I get a chance and we all agree it should happen?

**Comment by ronanyeah** [comment]
@m4v15 Yeah sounds good.

File: README.md [hunk 1]
```diff
@@ -6,116 +6,4 @@ A master reference for the running of [Founders &amp; Coders](http://www.founder
...
-#### Workshops
-* [Introduction to Express JS workshop](https://github.com/foundersandcoders/introduction-to-express)
-* [Express Code-Along workshop](https://github.com/foundersandcoders/express-workshop)
-* [Express Handlebars workshop](https://github.com/foundersandcoders/express-handlebars-workshop)
-* [Express and testing workshop](https://github.com/foundersandcoders/express-and-testing-workshop)
-
-#### Morning challenges
-* [Making an animated App drawer challenge](https://github.com/foundersandcoders/morning-challenge-animated-app-drawer)
-* [Express handlebars Challenge](https://github.com/foundersandcoders/express-handlebars-challenge)
+[ INSERT FILES + DIRECTORY STRUCTURE EXPLANATION HERE ]
```
Comment (eliascodes) [inline README.md]: Is this going to be done as part of this PR?

Comment (ronanyeah) [reply README.md]: Should probably just be an issue, I'll remove it.
https://github.com/foundersandcoders/master-reference/issues/746

**Pull request merged**